synthtool/gcp/snippets.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import re
+from typing import Dict, List
+
+OPEN_SNIPPET_REGEX = r".*\[START ([a-z0-9_]+)\].*$"
+CLOSE_SNIPPET_REGEX = r".*\[END ([a-z0-9_]+)\].*$"
```

Comment:
What about exclude regions like this one?
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/6879d73123135459fa45a36211505d1ead37978a/appengine/standard/endpoints-frameworks-v2/echo/app.yaml#L7

Instead of re-implementing the logic from the doc tag parser, can we re-use the same code as the doc tag parser?

Reply:
I didn't know that exclude tag was a thing, however it looks like a custom, per-widget regex supplied in the content template. We also haven't open-sourced the framework for cloud.google.com and that logic is included there.